e2e/cypress/common/permissions/main.ts

```diff
@@ -0,0 +1,80 @@
+import Bluebird from 'cypress/types/bluebird';
+import { login } from '../apiCalls/common';
+import {
+  generatePermissionsData,
+  PermissionsOptions,
+} from '../apiCalls/testData/testData';
+import { visitProjectDashboard } from '../shared';
+import { selectLangsInLocalstorage } from '../translations';
+import { testDashboard } from './dashboard';
+import { testKeys } from './keys';
+import { getProjectInfo, pageIsPermitted, ProjectInfo } from './shared';
+import { testTranslations } from './translations';
+
+export const SKIP = false;
+export const RUN = true;
+
+export function checkNumberOfMenuItems(count: number) {
+  cy.gcy('project-menu-items')
+    .findDcy('project-menu-item')
+    .should('have.length', count);
+}
+
+type MenuItem = Exclude<
+  DataCy.Value & `project-menu-item-${string}`,
+  'project-menu-item-projects'
+>;
+
+type Settings = Partial<Record<MenuItem, boolean>>;
+
+export function checkItemsInMenu(projectInfo: ProjectInfo, settings: Settings) {
+  checkNumberOfMenuItems(Object.keys(settings).length + 1);
```

Comment:
And what if by mistake, there is additional element and missing element in the same time. It should check the items separately.